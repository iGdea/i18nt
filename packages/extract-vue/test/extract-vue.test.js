const { describe, it, expect } = require('@jest/globals');
const { extractVue } = require('../');

describe('#extract', () => {
  it('#base', () => {
    const fileName = '/root/example.vue';
    const fileContent = `
    <template>
      <div
        :xx1="i18n('i18n arr1')"
        :xx2="i18n('i18n arr2', 'subkey1')"
        :xx3="i18n.t({ subkey: 'subkey2' })\`i18n.t arr3\`"
      >
        {{i18n('i18n text1')}}
        {{i18n('i18n text2', 'subkey1')}}
        {{i18n.t({ subkey: 'subkey2' })\`i18n.t text3\`}}
      </div>
    </template>
    <script>
      i18n('i18n script1');
      i18n('');
      i18n(\`\`);
      i18n('i18n script2', 'subkey1');
      i18n('i18n script3', { language: 'en', subkey: 'subkey2' });
      i18n.jsEncode('jsEncode script4', [i18n('sub i18n')]);
    </script>
    `;

    const result = extractVue({
      fileName,
      fileContent,
    });
    expect(result).toMatchSnapshot();
  });
});
