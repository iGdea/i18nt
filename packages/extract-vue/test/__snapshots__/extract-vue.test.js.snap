// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#extract #base 1`] = `
{
  "list": [
    {
      "subkey": undefined,
      "text": "i18n text1",
    },
    {
      "subkey": "subkey1",
      "text": "i18n text2",
    },
    {
      "subkey": "subkey2",
      "text": "i18n.t text3",
    },
    {
      "subkey": undefined,
      "text": "i18n arr1",
    },
    {
      "subkey": "subkey1",
      "text": "i18n arr2",
    },
    {
      "subkey": "subkey2",
      "text": "i18n.t arr3",
    },
    {
      "subkey": undefined,
      "text": "i18n script1",
    },
    {
      "subkey": "subkey1",
      "text": "i18n script2",
    },
    {
      "subkey": "subkey2",
      "text": "i18n script3",
    },
    {
      "subkey": undefined,
      "text": "jsEncode script4",
    },
    {
      "subkey": undefined,
      "text": "sub i18n",
    },
  ],
}
`;
